contracts/RandomBeaconHistory.cdc

```diff
@@ -26,6 +26,19 @@ access(all) contract RandomBeaconHistory {
     /// The path of the Heartbeat resource in the deployment account
     access(all) let HeartbeatStoragePath: StoragePath
 
+    // Event emitted when missing SoRs from past heartbeats are detected and will be backfilled:
+    //  - `blockHeight` is the height where the gap is detected
+    //  - `gapStartHeight` is the height of the first missing entry detected
+    access(all) event RandomHistoryMissing(blockHeight: UInt64, gapStartHeight: UInt64)
+
+    // Event emitted when missing SoRs are backfilled on the current heartbeat:
+    //  - `blockHeight` is the height where the backfill happened, it also defines the SoR used to backfill
+    //  - `gapStartHeight` is the height of the first backfilled entry
+    //  - `count` is the number of backfilled entries
+    // Note that in very rare cases, the backfilled gap may not be contiguous. This event does not
```

Comment:
The reason for adding 2 events and not only `RandomHistoryBackfilled` is the edge case specifically mentioned in the comment. 
If that edge case happens while only  emitting `RandomHistoryBackfilled`, there is no clear way to know what exact indices got backfilled.

Reply:
I copied the suggested event definition from https://github.com/bluesign/flow-core-contracts/blob/randomBackfill/contracts/RandomBeaconHistory.cdc